app/src/main/java/agency/five/codebase/android/movieapp/ui/favorites/FavoritesViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package agency.five.codebase.android.movieapp.ui.favorites
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.ui.favorites.mapper.FavoritesMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class FavoritesViewModel(
+    private val movieRepository: MovieRepository, private val favoritesMapper: FavoritesMapper
+) : ViewModel() {
+
+    private val initialViewState = FavoritesViewState(emptyList())
+
+    private val _favoritesViewState = MutableStateFlow(initialViewState)
+    val favoritesViewState = _favoritesViewState.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            movieRepository.favoriteMovies().collect(){ movies ->
+                _favoritesViewState.value = favoritesMapper.toFavoritesViewState(movies)
+            }
```

Comment:
Instead of doing this in `init` we can do it directly during variable intialization.
`val favoritesViewState: StateFlow<...> = movieRepository.favoriteMovies().map(...).stateIn(...)`

Reply:
funny thing is that I actually started with that (you can see in previous commit where I started implementing viewmodels) but didn't quite understand how it works + thought it kind of looked ugly so I switched to more familiar way with init block. :)